docs/guides/usage.md

```diff
@@ -0,0 +1,158 @@
+
+```python
+# structure:
+#   dir/a.txt
+#      /b.yaml
+#      /c.csv
+#      /...
+
+fs.put("dir", "my-repo/my-ref/dir", recursive=True)
+```
+
+!!! info
+
+    The above method of file uploading results in two transfers: Once from the client to the lakeFS server, and once from the lakeFS server to the object storage.
+    This can impact performance if the uploaded files are very large. To avoid this performance hit, you can also decide to write the file directly to the underlying object storage:
+
+    ```python
+    fs = LakeFSFileSystem()
+    
+    fs.put_file("my-repo/my-ref/file.txt", "file.txt", use_blockstore=True)
+    ```
+
+    Direct lakeFS blockstore uploads require the installation of the corresponding `fsspec` file system implementation through `pip`.
+    For an S3-based lakeFS deployment, install the `s3fs` package. For Google Cloud Storage (GCS), install the `gcsfs` package.
+    For Azure blob storage, install the `adlfs` package.
+
+### File downloads
+
+To download a file, you can use the `fs.get()` or `fs.get_file()` methods:
+
+```python
+from lakefs_spec import LakeFSFileSystem
+
+fs = LakeFSFileSystem()
+
+# remote path, then local target path.
+fs.get_file("my-repo/my-ref/file.txt", "file.txt")
+```
+
+In the case of a directory in lakeFS, use the `fs.get()` API together with the `recursive=True` switch:
+
+```python
+# structure:
+#   dir/a.txt
+#      /b.yaml
+#      /c.csv
+#      /...
+
+# downloads `dir` into the current directory.
+fs.get("my-repo/my-ref/dir", "dir", recursive=True)
+```
+
+## Obtaining info on stored objects
+
+To query (also called `stat`) a single object in a lakeFS repository, use the `fs.info()` API:
+
+```python
+from lakefs_spec import LakeFSFileSystem
+
+fs = LakeFSFileSystem()
+
+my_file_info = fs.info("my-repo/my-ref/my-file.txt")
+```
+
+The resulting `my_file_info` object is a dictionary containing useful information such as storage location of the file, creation timestamp, and size (in bytes).
+
+You can also call `fs.info()` on directories:
+
+```python
+dir_info = fs.info("my-repo/my-ref/dir/")
+```
+
+In this case, the resulting `dir_info` object only contains the directory name, and the cumulated size of the files it contains.
+
+## Listing directories in lakeFS
+
+To list the files in a directory in lakeFS, use the `fs.ls()` method:
+
+```python
+from lakefs_spec import LakeFSFileSystem
+
+fs = LakeFSFileSystem()
+
+my_dir_listing = fs.ls("my-repo/my-ref/my-dir/")
+```
+
+This returns a list of Python dictionaries containing information on the objects contained in the requested directory.
+The returned objects have the same fields set as those returned by a normal `fs.info()` call on a file object.
+
+## Deleting objects from a lakeFS branch
+
+To delete objects from a lakeFS branch, use the `fs.rm_file()` or `fs.rm()` APIs. As before, while the former works only for single files, the latter can be used to remove whole directories with the `recursive=True` option.
```

Comment:
```suggestion
To delete objects from a lakeFS branch, use the `fs.rm_file()` or `fs.rm()` APIs. As before, while the former works only for single files, the latter can be used to remove entire directories with the `recursive=True` option.
```

Reply:
Also fixed in [13917c0](https://github.com/aai-institute/lakefs-spec/pull/156/commits/13917c064afc3816ce4cca9787ebb3ad784d3c98).